File: leetcode/assign-cookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    g.sort((a, b) => a - b);
    s.sort((a, b) => a - b);
    
    let cursor = 0;
    let r = 0;
    for(let i = 0; i< s.length; i++) {
        if(g[cursor] <= s[i]){
            r++;
            cursor++;
        }
    }
    
    return r;
};

File: leetcode/assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        grids = sorted(g)
        cookies = sorted(s)
        
        ans = 0
        cursor = 0
        
        for cookie in cookies:
            if grids[cursor] <= cookie:
                ans += 1
                cursor += 1;
            
            if cursor == len(grids):
                break;
                
        return ans
